x/stake/keeper/validator.go

```diff
@@ -452,21 +453,32 @@ func (k Keeper) UpdateBondedValidators(
 	// swap the cliff validator for a new validator if the affected validator
 	// was bonded
 	if newValidatorBonded {
-		// unbond the cliff validator
 		if oldCliffValidatorAddr != nil {
-			cliffVal, found := k.GetValidator(ctx, oldCliffValidatorAddr)
+			oldCliffVal, found := k.GetValidator(ctx, oldCliffValidatorAddr)
 			if !found {
 				panic(fmt.Sprintf("validator record not found for address: %v\n", oldCliffValidatorAddr))
 			}
 
-			k.unbondValidator(ctx, cliffVal)
+			affectedValRank := GetValidatorsByPowerIndexKey(affectedValidator, pool)
+			valRank := GetValidatorsByPowerIndexKey(validator, pool)
+
+			if bytes.Compare(valRank, affectedValRank) < 0 {
+				// unbond the old cliff validator iff the affected validator was
+				// newly bonded and has greater power
+				k.unbondValidator(ctx, oldCliffVal)
+			} else {
+				// otherwise unbond the affected validator, which must have been
+				// kicked out
+				affectedValidator = k.unbondValidator(ctx, affectedValidator)
+			}
 		}
 
-		// bond the new validator
 		validator = k.bondValidator(ctx, validatorToBond)
 		if bytes.Equal(validator.Owner, affectedValidator.Owner) {
 			return validator, true
 		}
+
+		return affectedValidator, true
```

Comment:
Not sure if this is ideal, but in case line # 472 get executed, I need to make sure `UpdateValidator` set's the newly modified `affectedValidator` which should have it's status set to unbonded.

Reply:
This seems right